### ISSUE TITLE
Better error message for NoSuchKernel

### ISSUE THREAD

**Issue by TomAugspurger** [jupyter/jupyter_client#126]
Trying to run `nbconvert` on a notebook, and I get 

```python
  File "<snip>/site-packages/jupyter_client/kernelspec.py", line 143, in get_kernel_spec
    raise NoSuchKernel(kernel_name)
jupyter_client.kernelspec.NoSuchKernel
```

It'd be helpful if the message included the kernel it's looking for and the set of known kernels.

Sidenote, does `nbconvert` have a 4.2 release? I'm trying to pass the `kernel_name` argument [here](http://nbconvert.readthedocs.org/en/latest/execute_api.html), but I see that I'm on `4.1` (off PyPI) and that argument was added in 4.2.

An example

```python
In [7]: from jupyter_client import KernelManager

In [8]: manager = KernelManager()

In [9]: manager.kernel_spec_manager.get_kernel_spec('foo')
---------------------------------------------------------------------------
NoSuchKernel                              Traceback (most recent call last)
<ipython-input-9-ff3677045dc1> in <module>()
----> 1 manager.kernel_spec_manager.get_kernel_spec('foo')

/Users/tom.augspurger/Envs/py27/lib/python2.7/site-packages/jupyter_client/kernelspec.pyc in get_kernel_spec(self, kernel_name)
    141             resource_dir = d[kernel_name.lower()]
    142         except KeyError:
--> 143             raise NoSuchKernel(kernel_name)
    144
    145         if kernel_name == NATIVE_KERNEL_NAME:

NoSuchKernel:
```

**Comment by TomAugspurger** [comment]
This one seems very sprint-friendly FYI; would be a nice one to fix during Jupyter day Chicago this weekend :wink:

**Comment by takluyver** [comment]
Would be sprint friendly, but it was already fixed in #108.

4.1.0 is the current stable release.

**Comment by TomAugspurger** [comment]
Thanks! Sorry I missed that.
